src/test/java/seedu/address/logic/commands/EditTaskCommandTest.java

```diff
@@ -0,0 +1,112 @@
+import seedu.address.testutil.TypicalTasks;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for EditTaskCommand.
+ */
+public class EditTaskCommandTest {
+
+    private AddressBook typicalPersonsAddressBookWithTask = TypicalTasks
+            .addTypicalTasksToAddressBook(new AddressBook());
+    private Model model = new ModelManager(typicalPersonsAddressBookWithTask, new UserPrefs());
+
+    @Test
+    public void execute_allFieldsSpecifiedUnfilteredList_success() {
+        Task editedTask = new TaskBuilder().build();
+        EditTaskDescriptor descriptor = new EditTaskDescriptorBuilder(editedTask).build();
+        EditTaskCommand editTaskCommand = new EditTaskCommand(INDEX_FIRST_TASK, descriptor);
+
+        String expectedMessage = String.format(EditTaskCommand.MESSAGE_EDIT_TASK_SUCCESS, editedTask);
+
+        Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.setTask(model.getFilteredTaskList().get(0), editedTask);
+
+        assertCommandSuccess(editTaskCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_someFieldsSpecifiedUnfilteredList_success() {
+        Index indexLastTask = Index.fromOneBased(model.getFilteredTaskList().size());
+        Task lastTask = model.getFilteredTaskList().get(indexLastTask.getZeroBased());
+
+        TaskBuilder taskInList = new TaskBuilder(lastTask);
+        Task editedTask = taskInList.withLabel(VALID_LABEL_ORDER).build();
+
+        EditTaskDescriptor descriptor = new EditTaskDescriptorBuilder().withLabel(VALID_LABEL_ORDER).build();
+        EditTaskCommand editTaskCommand = new EditTaskCommand(indexLastTask, descriptor);
+
+        String expectedMessage = String.format(EditTaskCommand.MESSAGE_EDIT_TASK_SUCCESS, editedTask);
+
+        Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.setTask(lastTask, editedTask);
+
+        assertCommandSuccess(editTaskCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_duplicateTaskUnfilteredList_failure() {
+        Task firstTask = model.getFilteredTaskList().get(INDEX_FIRST_TASK.getZeroBased());
+        EditTaskDescriptor descriptor = new EditTaskDescriptorBuilder(firstTask).build();
+        EditTaskCommand editTaskCommand = new EditTaskCommand(INDEX_SECOND_TASK, descriptor);
+
+        assertCommandFailure(editTaskCommand, model, MESSAGE_DUPLICATE_TASK);
+    }
+
+    @Test
+    public void execute_noChangesMadeUnfilteredList_failure() {
+        Task firstTask = model.getFilteredTaskList().get(INDEX_FIRST_TASK.getZeroBased());
+        EditTaskDescriptor descriptor = new EditTaskDescriptorBuilder(firstTask).build();
+        EditTaskCommand editTaskCommand = new EditTaskCommand(INDEX_FIRST_TASK, descriptor);
+
+        assertCommandFailure(editTaskCommand, model, MESSAGE_NO_CHANGES_MADE);
+    }
+
```

Comment:
Is it better to test out-of-bound scenarios too?
e.g. ```public void execute_invalidTaskIndexUnfilteredList_failure()```

Reply:
Nice spot! I've added the test case